tidy3d/material_library/parametric_materials.py

```diff
@@ -0,0 +1,266 @@
+            A pole-residue model for the interband term of graphene.
+        """
+
+        mu_c_freq = self.mu_c / (np.pi * HBAR)
+        temp_freq = K_B * self.temp / (np.pi * HBAR)
+
+        freq1 = mu_c_freq
+        freq2 = mu_c_freq + 5 * temp_freq
+
+        ind1 = int(
+            NUM_FREQS * (freq1 - MIN_FREQ_THZ * 1e12) / (MAX_FREQ_THZ * 1e12 - MIN_FREQ_THZ * 1e12)
+        )
+        ind2 = int(
+            NUM_FREQS * (freq2 - MIN_FREQ_THZ * 1e12) / (MAX_FREQ_THZ * 1e12 - MIN_FREQ_THZ * 1e12)
+        )
+        inds = np.array([[0, -1], [ind1, ind2]])
+
+        omega = 2 * np.pi * np.linspace(MIN_FREQ_THZ * 1e12, MAX_FREQ_THZ * 1e12, NUM_FREQS)
+        sigma_inter = self.interband_conductivity(omega)
+
+        return self._fit_interband_conductivity(omega, sigma_inter, inds)
+
+    def interband_conductivity(self, omegas: List[float]) -> List[complex]:
+        """Numerically integrate interband term."""
+
+        def fermi(E: float) -> float:
+            """Fermi distribution."""
+            return 1 / (np.exp((E - self.mu_c) / (K_B * self.temp)) + 1)
+
+        def fermi_g(E: float) -> float:
+            """Difference of fermi distributions."""
+            return fermi(-E) - fermi(E)
+
+        inter = np.zeros(len(omegas), dtype=complex)
+
+        def integrand(E, omega):
+            return (fermi_g(E * HBAR) - fermi_g(HBAR * omega / 2)) / (omega**2 - 4 * E**2)
+
+        for i, omega in enumerate(omegas):
+
+            integral, _ = integrate.quad(integrand, 1e11, 1e16, args=(omega,), epsabs=1e-20)
+            inter[i] = (Q_e / (4 * HBAR * EPSILON_0)) * (
+                fermi_g(HBAR * omega / 2) - (4 * omega / (1j * np.pi)) * integral
+            )
+
+        return inter * EPSILON_0
+
+    def _fit_interband_conductivity(
+        self,
+        omega: List[float],
+        sigma: List[complex],
+        indslist: List[Tuple[int, int]],
+        num_iters: int = 100,
+    ):
+        """Fit the interband conductivity with a Pade approximation, as described in
+
+        Stamatios Amanatiadis, Theodoros Zygiridis, Tadao Ohtani, Yasushi Kanai,
+        and Nikolaos Kantartzis, "A Consistent Scheme for the Precise FDTD Modeling
+        of the Graphene Interband Contribution," IEEE Trans. Magn. 57, 6 (2021).
+
+        Parameters
+        ----------
+        omega : List[float]
+            The input of angular frequencies.
+        sigma : List[complex]
+            The interband conductivity to fit.
+        indslist : List[Tuple[int, int]]
+            The indices at which to sample the data for fitting.
+            The length of this list determines the number of Pade terms used.
+        num_iters : int
+            The number of optimization iterations.
+        Returns
+        -------
+        :class:`.PoleResidue`
+            A pole-residue model approximating the interband conductivity.
+        """
+
+        def evaluate(omega: List[float], coeffslist: List[List[float]]) -> List[float]:
```

Comment:
Might be better to have a more descriptive name. `evaluate_coefficients`?
Do we know `coeffslist` will contain exactly 4 lists of coeffs?

Reply:
Each item in coeffslist is a list of exactly four floats. I added a comment to this effect.

Reply:
oh right. great, thanks. Can also do `List[Tuple[float, float, float, float]]` but it might take up a bit of room so up to you.